core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -230,6 +233,7 @@ public DefaultCacheManager(GlobalConfiguration globalConfiguration, Configuratio
       this.globalComponentRegistry.registerComponent(cacheDependencyGraph, CACHE_DEPENDENCY_GRAPH, false);
       this.globalComponentRegistry.registerComponent(authzHelper, AuthorizationHelper.class);
       this.stats = new CacheContainerStatsImpl(this);
+      this.health = HealthFactory.create(this);
```

Comment:
If the `Health` instance is always created from the `DefaultCacheManager` then I don't see a need for the factory.